Mathlib/Data/Complex/Basic.lean

```diff
@@ -342,39 +370,65 @@ defined in `data.complex.module.lean`. -/
 instance : Nontrivial ℂ :=
   pullback_nonzero re rfl rfl
 
-instance : AddCommGroup ℂ := by
-  refine_struct { zero := (0 : ℂ)
-                  add := (· + ·)
-                  neg := Neg.neg
-                  sub := Sub.sub
-                  nsmul := fun n z => ⟨n • z.re - 0 * z.im, n • z.im + 0 * z.re⟩
-                  zsmul := fun n z => ⟨n • z.re - 0 * z.im, n • z.im + 0 * z.re⟩ } <;> intros <;>
-            try rfl <;> apply ext_iff.2 <;> constructor <;> simp <;> · first |ring1|ring_nf
-
-instance : AddGroupWithOne ℂ :=
+-- Porting note: proof needed modifications and rewritten fields
+instance addCommGroup : AddCommGroup ℂ := {
+  zero := (0 : ℂ)
+  add := (· + ·)
+  neg := Neg.neg
+  sub := Sub.sub
+  nsmul := fun n z => ⟨n • z.re - 0 * z.im, n • z.im + 0 * z.re⟩
+  zsmul := fun n z => ⟨n • z.re - 0 * z.im, n • z.im + 0 * z.re⟩
+  zsmul_zero':= by intros ; ext <;> simp
+  nsmul_zero := by intros ; ext <;> simp
+  nsmul_succ := by
+    intros ; ext <;> simp [AddMonoid.nsmul_succ, add_mul, add_comm]
+  zsmul_succ' := by
+    intros; ext <;> simp [SubNegMonoid.zsmul_succ', add_mul, add_comm]
+  zsmul_neg' := by
+    intros ; ext <;> simp [zsmul_neg', add_mul]
+  add_assoc := by intros ; ext <;> simp [add_assoc]
+  zero_add := by intros ; ext <;> simp
+  add_zero := by intros ; ext <;> simp
+  add_comm := by intros ; ext <;> simp [add_comm]
+  add_left_neg := by intros ; ext <;> simp }
+
+
+instance Complex.addGroupWithOne : AddGroupWithOne ℂ :=
   { Complex.addCommGroup with
     natCast := fun n => ⟨n, 0⟩
-    natCast_zero := by ext <;> simp [Nat.cast]
-    natCast_succ := fun _ => by ext <;> simp [Nat.cast]
+    natCast_zero := by
+      ext <;> simp [Nat.cast, AddMonoidWithOne.natCast_zero]
+    natCast_succ := fun _ => by ext <;> simp [Nat.cast, AddMonoidWithOne.natCast_succ]
     intCast := fun n => ⟨n, 0⟩
-    intCast_ofNat := fun _ => by ext <;> simp [fun n => show @coe ℕ ℂ ⟨_⟩ n = ⟨n, 0⟩ from rfl]
-    intCast_negSucc := fun _ => by ext <;> simp [fun n => show @coe ℕ ℂ ⟨_⟩ n = ⟨n, 0⟩ from rfl]
+    intCast_ofNat := fun _ => by ext <;> rfl
+    intCast_negSucc := fun n => by
+      ext
+      · simp [AddGroupWithOne.intCast_negSucc]
+        show -(1: ℝ) + (-n) = -(↑(n + 1))
+        simp [Nat.cast_add, add_comm]
+      · simp [AddGroupWithOne.intCast_negSucc]
+        show im ⟨n, 0⟩ = 0
+        rfl
     one := 1 }
 
-instance : CommRing ℂ := by
-  refine_struct
-                { Complex.addGroupWithOne with
-                  zero := (0 : ℂ)
-                  add := (· + ·)
-                  one := 1
-                  mul := (· * ·)
-                  npow := @npowRec _ ⟨(1 : ℂ)⟩ ⟨(· * ·)⟩ } <;>
-              intros <;>
-            try rfl <;>
-          apply ext_iff.2 <;>
-        constructor <;>
-      simp <;>
-    · first |ring1|ring_nf
+instance commRing : CommRing ℂ :=
```

Comment:
This should probably have a porting note too, since the instance above does.

Reply:
Added